stratum/hal/lib/tdi/tdi_sde_helpers.cc

```diff
@@ -200,15 +207,16 @@ ::util::Status GetFieldExact(const ::tdi::TableKey& table_key,
   const ::tdi::Table* table;
   tdi_field_data_type_e data_type;
   RETURN_IF_TDI_ERROR(table_key.tableGet(&table));
-  ::tdi::KeyFieldValueExact<uint64_t> key_field_value(*field_value);
+  ::tdi::KeyFieldValueExact<int64_t> key_field_value(*field_value);
   const ::tdi::KeyFieldInfo* keyFieldInfo =
       table->tableInfoGet()->keyFieldGet(field_name);
   RETURN_IF_NULL(keyFieldInfo);
 
   field_id = keyFieldInfo->idGet();
   data_type = keyFieldInfo->dataTypeGet();
 
-  RET_CHECK(data_type == TDI_FIELD_DATA_TYPE_UINT64)
+  RET_CHECK(data_type == TDI_FIELD_DATA_TYPE_UINT64 ||
+	   data_type == TDI_FIELD_DATA_TYPE_BYTE_STREAM)
       << "Requested uint64 but field " << field_name << " has type "
```

Comment:
yeah byte stream probably needs slightly better handling considering cases where data width>64 bits. This will have to go somewhere in common code since bytes are in TDI common interface.